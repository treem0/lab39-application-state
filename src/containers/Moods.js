import React from 'react';
import Controls from '../components/controls/Controls';
import Face from '../components/face/Face';
import { useDispatch, useSelector } from 'react-redux';
import { drinkCoffee, eatSnack, takeNap, study } from '../actions/emojiActions';
import { getFace } from '../selectors/emojiSelector';


const Moods = () => {
  const dispatch = useDispatch();
  const emojiFace = useSelector(getFace);

  return (
    <>
      <Controls>
        <button onClick={() => dispatch(drinkCoffee())}>coffee</button>
        <button onClick={() => dispatch(eatSnack())}>snacks</button>
        <button onClick={() => dispatch(takeNap())}>naps</button>
        <button onClick={() => dispatch(study())}>studies</button>
      </Controls>
      <Face emoji={emojiFace} />
    </>
  );
};

export default Moods;
