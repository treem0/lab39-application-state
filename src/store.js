const { createStore } = require('redux');
import reducer from './reducers/emojiReducer';

export default createStore(
  reducer,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()

);
