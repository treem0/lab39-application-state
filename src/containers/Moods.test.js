import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Moods from './Moods';
import store from '../store';
describe('container', () => {
  it('matches a snapshot', () => {
    const wrapper = mount(<Provider store={store}><Moods /></Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
