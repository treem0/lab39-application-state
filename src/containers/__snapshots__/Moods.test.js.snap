// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`container matches a snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Moods>
    <Controls>
      <section
        className="Controls"
      >
        <button
          onClick={[Function]}
        >
          coffee - 
          0
        </button>
        <button
          onClick={[Function]}
        >
          snacks - 
          0
        </button>
        <button
          onClick={[Function]}
        >
          naps - 
          0
        </button>
        <button
          onClick={[Function]}
        >
          studies - 
          0
        </button>
      </section>
    </Controls>
    <Face
      emoji="ðŸ¤¬"
    >
      <p
        className="Face"
      >
        ðŸ¤¬
      </p>
    </Face>
  </Moods>
</Provider>
`;
